src/gui/widgets/TabWidget.cpp

```diff
@@ -135,88 +200,104 @@ void TabWidget::resizeEvent( QResizeEvent * )
-							QPalette::Background );
-	QLinearGradient g( 0, 0, 0, m_tabheight );
-	g.setColorAt( 0, bg_color.darker( 250 ) );
-	g.setColorAt( 0.1, bg_color.lighter( 120 ) );
-	g.setColorAt( 1, bg_color.darker( 250 ) );
+	// Draw background
+	QBrush bg_color = p.background();
 	p.fillRect( 0, 0, width() - 1, height() - 1, bg_color );
 
-	bool big_tab_captions = ( m_caption == "" );
-
-	p.setPen( bg_color.darker( 150 ) );
+	// Draw external borders
+	p.setPen( tabBorder() );
 	p.drawRect( 0, 0, width() - 1, height() - 1 );
 
-	p.setPen( bg_color.light( 150 ) );
-	p.drawLine( width() - 1, 0, width() - 1, height() - 1 );
-	p.drawLine( 0, height() - 1, width() - 1, height() - 1 );
-
-	p.setPen( QColor( 0, 0, 0 ) );
-	p.drawRect( 1, 1, width() - 3, height() - 3 );
-
-	p.fillRect( 2, 2, width() - 4, m_tabheight, g );
-	p.drawLine( 2, m_tabheight + 2, width() - 3, m_tabheight + 2);
+	// Draw tabs' bar background
+	p.fillRect( 1, 1, width() - 2, m_tabheight + 2, tabBackground() );
 
+	// Draw title, if any
 	if( ! m_caption.isEmpty() )
 	{
-		p.setPen( QColor( 255, 255, 255 ) );
+		p.setFont( pointSize<8>( p.font() ) );
+		p.setPen( tabTitleText() );
 		p.drawText( 5, 11, m_caption );
 	}
 
-	// Calculate the tabs' x (tabs are painted next to the caption)
-	int tab_x_offset = m_caption.isEmpty() ? 4 : 14 + fontMetrics().width( m_caption );
+        // Calculate the tabs' x (tabs are painted next to the caption)
+        int tab_x_offset = m_caption.isEmpty() ? 4 : 14 + fontMetrics().width( m_caption );
 
-	QColor cap_col( 160, 160, 160 );
-	if( big_tab_captions )
+	// Compute tabs' width depending on the number of tabs (only applicable for artwork tabs)
+  	widgetStack::iterator first = m_widgets.begin();
+  	widgetStack::iterator last = m_widgets.end();
+	int tab_width = width();
+	if ( first != last )
 	{
-		p.setFont( pointSize<8>( p.font() ) );
-		cap_col = QColor( 224, 224, 224 );
-	}
-	else
-	{
-		p.setFont( pointSize<7>( p.font() ) );
+		tab_width = ( width() - tab_x_offset ) / std::distance( first, last );
 	}
-	p.setPen( cap_col );
 
-
-	for( widgetStack::iterator it = m_widgets.begin();
-						it != m_widgets.end(); ++it )
-	{
-		if( it.key() == m_activeTab )
+	// Draw all tabs
+        for( widgetStack::iterator it = first ; it != last ; ++it )
+        {
+
+		// Draw a text tab or a artwork tab.
+		if ( m_usePixmap )
+		{	
+			// Fixes tab's width, because original size is only correct for text tabs
+			( *it ).nwidth = tab_width;
+
+			// Create a mask out of the tab's artwork (for changing it's color later on)
+			QBitmap mask = QPixmap( embed::getIconPixmap( ( *it ).activePixmap ) ).createMaskFromColor( Qt::black, Qt::MaskOutColor );
+
+			// Select artwork's color and highlight active tab
+			if( it.key() == m_activeTab )
+                        {
+				p.fillRect( tab_x_offset, 0, ( *it ).nwidth, m_tabbarHeight - 1, tabSelected() );
+				p.setPen( tabArtworkActive() );
+			} else 
```

Comment:
For consistency use:

```
}
else
{
```

Per https://github.com/LMMS/lmms/wiki/Coding-conventions